test_unstructured/partition/pdf_image/test_image.py

```diff
@@ -737,3 +741,71 @@ def test_partition_image_works_on_heic_file(
     elements = image.partition_image(filename=filename, strategy=PartitionStrategy.AUTO)
     titles = [el.text for el in elements if el.category == ElementType.TITLE]
     assert "CREATURES" in titles
+
+
+@pytest.mark.parametrize(
+    ("strategy", "expected_ids_"),
+    [
+        (PartitionStrategy.HI_RES, "expected_element_ids_for_hi_res_strategy_"),
+        (PartitionStrategy.OCR_ONLY, "expected_element_ids_for_ocr_strategy_"),
+    ],
+    indirect=["expected_ids_"],
+)
+def test_deterministic_element_ids(strategy: str, expected_ids_: Any):
+    elements = image.partition_image(
+        example_doc_path("layout-parser-paper-with-table.jpg"),
+        strategy=strategy,
+        starting_page_number=2,
+    )
+    ids = [element.id for element in elements]
+    assert ids == expected_ids_, "Element IDs do not match expected IDs"
+
+
+# -- fixtures --------------------------------------------------------------------------------
+
+
+@pytest.fixture()
+def expected_ids_(request: SubRequest):
+    return request.getfixturevalue(request.param)
+
+
+@pytest.fixture()
+def expected_element_ids_for_hi_res_strategy_():
+    """Expected ids with strategy `hi_res` for 'layout-parser-paper-with-table.jpg'."""
+    return [
+        "34cbf83e5f9eb1527247ab062819b1ca",
+        "347d549be4deab25a3ee4f485f831c7d",
+        "3eab2e82aef0b92793e3c68716f54aa2",
+        "56ed2d576905d1d4907956e41be91866",
+        "7ae9174d036f1d261cc4c8dfc1b7d9c9",
+        "78a9454bf9fa17d79d2b28b919e0937b",
+        "9f87d002da6a94daec6a1b957b2fedfe",
+        "8211d66d55646be7f798172ea5596979",
+        "aa7c780f8442901becaf1c4f7067708e",
+        "9516561be9106b4867dd7bfeeafa895d",
+        "77b2a946442eef307412d62f284757ff",
+        "e557fb5a2472f5c9be22e6bd07cc6b7a",
+        "23d5472327ea19d8331cd5b0ed970e3f",
+        "6c67796c499a68160c107006a99a9392",
+        "7468807f52b225b01404f9f3d7c9e6e2",
+    ]
```

Comment:
i know we have this pattern elsewhere in the tests, but it is not great to manually have to update these when the ID hashing algo changes.
instead, let's run partition twice (even though the test will take longer) and make sure we are getting back the same element_id's.